src/mds/MDSRank.cc

```diff
@@ -2983,11 +2983,71 @@ bool MDSRankDispatcher::handle_command(
 
     damage_table.erase(id);
     return true;
+  } else if (prefix == "drop cache") {
+    command_drop_cache(ds);
+    return true;
   } else {
     return false;
   }
 }
 
+void MDSRank::command_drop_cache(stringstream *ds)
+{
```

Comment:
add `Mutex::Locker l(mds_lock);`

Reply:
@batrick Using `Mutex::Locker l(mds_lock);` creates the same issue as before [1]. At least for time being, I am replacing it by `mds_lock.Lock()` so that MDS would stay locked until control reaches here [2] after exiting `command_drop_cache()`; this prevents MDS from crashing. 

[1] https://github.com/ceph/ceph/pull/21566#discussion_r184181266
[2] https://github.com/ceph/ceph/blob/master/src/mds/MDSDaemon.cc#L1143

Reply:
Oh, right that lock is not necessary because it's locked in `MDSDaemon::ms_dispatch`.